data-processor/src/main/java/io/micronaut/data/processor/visitors/finders/AbstractCriteriaMethodMatch.java

```diff
@@ -556,6 +556,12 @@ private boolean isValidType(ClassElement genericType, SourcePersistentProperty p
         if (TypeUtils.areTypesCompatible(genericType, property.getType()) && !TypeUtils.isObjectClass(genericType)) {
             return true;
         }
+        if (TypeUtils.isContainerType(property.getType())) {
```

Comment:
Without this fix, property tags can be array of `String` but not `List<String>` so I think it makes sense checking this too. It did not break anything else so should be safe.